homeassistant/components/duotecno/__init__.py

```diff
@@ -0,0 +1,39 @@
+"""The duotecno integration."""
+from __future__ import annotations
+
+from duotecno.controller import PyDuotecno
+from duotecno.exceptions import InvallidPassword, LoadFailure
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_PORT, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import PlatformNotReady
+
+from .const import DOMAIN
+
+PLATFORMS: list[Platform] = [Platform.SWITCH]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up duotecno from a config entry."""
+
+    hass.data.setdefault(DOMAIN, {})
+    controller = PyDuotecno()
+    hass.data[DOMAIN][entry.entry_id] = controller
```

Comment:
Let's not set this before we actually are connected, so move it down a bit (after the try block).